### DISCUSSION TITLE
When there are conflicting classes, how to make the later ones overwrite the style of the previous ones?

### DISCUSSION THREAD

**Discussion by lihanspace** [tailwindlabs/tailwindcss#11105] [Ideas]
When using a Vue component, if I want to override its style, writing like this will not work.

```vue
<!-- Test.vue -->
<template>
  <div class="bg-[red] w-[200px] h-[200px]"></div>
</template>
```

```vue
<!-- App.vue -->
...
<Test class="bg-[green] w-[100px] h-[100px]" />
...
```

```html
<!-- Built index.html file -->
...
<style>
  .h-\[100px\] {
    height: 100px;
  }
  .h-\[200px\] {
    height: 200px;
  }
  .w-\[100px\] {
    width: 100px;
  }
  .w-\[200px\] {
    width: 200px;
  }
  .bg-\[green\] {
    --tw-bg-opacity: 1;
    background-color: rgb(0 128 0 / var(--tw-bg-opacity));
  }
  .bg-\[red\] {
    --tw-bg-opacity: 1;
    background-color: rgb(255 0 0 / var(--tw-bg-opacity));
  }
</style>
...
<div class="bg-[red] w-[200px] h-[200px] bg-[green] w-[100px] h-[100px]"></div>
...
```

![image](https://user-images.githubusercontent.com/60546798/234751335-a2607cca-426a-4ce4-8514-1f12bdfef591.png)

How to make the later class overwrite the style of the previous class?

https://play.tailwindcss.com/aeWAv7gbfp

**Comment by wongjn** [comment]
You'd want to rework your templating such that only one of each category of class exists at any one time:

```html
<div class="bg-[red] w-[200px] h-[200px]"></div>
```
or
```html
<div class="bg-[green] w-[100px] h-[100px]"></div>
```

As a last-ditch attempt, you could use [the important prefix `!`](https://tailwindcss.com/docs/configuration#important-modifier) but I'd like to stress, that one should not really use this method at all:

```html
<div class="bg-[red] w-[200px] h-[200px] !bg-[green] !w-[100px] !h-[100px]"></div>
```
